src/lib/Gen/hbs/HandlebarsGenerator.cpp

```diff
@@ -15,13 +15,73 @@
 #include "Builder.hpp"
 #include "MultiPageVisitor.hpp"
 #include "SinglePageVisitor.hpp"
+
+#include "lib/Lib/TagfileWriter.hpp"
+#include "lib/Support/RawOstream.hpp"
+
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+
 #include <mrdocs/Support/Path.hpp>
+
+#include <fstream>
 #include <sstream>
 
 namespace clang {
 namespace mrdocs {
 namespace hbs {
 
+namespace {
+
+/** Generate a Tagfile for cross-referencing. */
```

Comment:
Minor grammar issue: "cross-referencing" needs a subject.

Reply:
What should `generatedFilesPath` represent in the single and multi-page cases? (Just the output path, if I understand correctly).

And why does the generator need this information? It doesn't need to know where these absolute files are because tag files only include relative paths available from the hbsCorpus.